views/_notes.erb

```diff
@@ -2,19 +2,53 @@
 
 <% unless notes.empty? %>
   <h3>Notes</h3>
-    <% notes.each do |note| %>
-      <div class="box">
-        <div class="columns is-full">
-          <div class="column">
-            <h4 class="mb-0"><%= note.body %></h4>
-            <small>Created at: <%= note.created_at %></small>
+
+  <% notes.each do |note| %>
+    <div class="box p-0" style="border: 1px solid #1c1c39">
+      <div class="columns m-0 p-1" style="background-color: #1c1c39">
+        <div class="column is-half px-3 py-0">
+          <small class="is-size-7">Created at: <%= note.created_at %></small>
+        </div>
+
+        <div class="column has-text-right p-0 pr-1 mt-0">
+          <div class="dropdown three-dots is-right">
+            <div class="dropdown-trigger">
+              <span class="icon is-small">
+                <i class="fas fa-ellipsis-h"></i>
+              </span>
+            </div>
+            <div class="dropdown-menu" id="dropdown-menu6" role="menu">
+              <div class="dropdown-content">
+                <a class="dropdown-item py-2 edit-note" data-note-id=<%= note.id %>>Edit</a>
+                <a class="dropdown-item py-2 delete-note has-text-danger" data-note-id=<%= note.id %>>Delete</a>
+              </div>
+            </div>
           </div>
-          <div class="column is-one-fifth has-text-right">
-            <button class="button is-danger is-small delete-note " data-note-id=<%= note.id %>>X</button>
+        </div>
+      </div>
+
+      <div class="columns is-full">
+        <div class="column">
+          <div id="note-edit-body-<%= note.id %>" class="mb-0 p-4"><%= note.body %></div>
```

Comment:
Probably a good idea to escape `note.id` with `hattr` just to be safe, and that it doesn't trigger false positives on some security scan.